Hw_7_Sotnikov.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+from math import ceil
+
+
+# # Pandas intersect
+
+
+# Functions for reading .GFF and .BED files
+
+def read_gff(path_to_file):
+    gff = pd.read_csv(path_to_file, sep="\t", names=["chrom", "source", "type", "start", "end", "score", "strand", "phase", "attributes"], header=0)
+    return gff
+
+def read_bed(path_to_file):
+    bed = pd.read_csv(path_to_file, sep="\t", names=["chrom", "chromStart", "chromEnd", "name", "bed_score", "bed_strand"])
+    return bed
+
+
+
+# Example of standard reading of .GFF file. Cutting long string around type of RNA
+
+rgff = read_gff("/home/asklepiad/bioinf/python/git_projects_BI_2022/hw6/rrna_annotation.gff")
+rgff["attributes"] = rgff["attributes"].str.removeprefix("Name=").str.replace("_rRNA.*", "")
+rgff
+
+
+# Table of RNA types number per chromosome
+
+number_of_rnas = rgff.groupby(["chrom", "attributes"], as_index=False).size()
+number_of_rnas["chrom"] = number_of_rnas["chrom"].str.removeprefix("Reference_")
+number_of_rnas
+
+
+# Creating barplot by previously-described data
+
+sns.barplot(number_of_rnas, x=number_of_rnas.iloc[:, 0], y=number_of_rnas.iloc[:, 2], hue=number_of_rnas.iloc[:, 1])
+
+
+# Reading .BED file
+
+rbed = read_bed("/home/asklepiad/bioinf/python/git_projects_BI_2022/hw6/alignment.bed")
+rbed
+
+
+# Bedtools intersect without bedtools
+
+inter_before = pd.merge(rgff, rbed, on="chrom")
```

Comment:
Это не совсем то. Когда ты замерджил датафреймы по хромосоме, то у тебя получились для каждой хромосомы все возможные комбинации записей из первого и второго датафрейма, но не все из них являются искомыми пересечениями.
```suggestion
inter_before = pd.merge(rgff, rbed, on="chrom")
intersection = inter_before.query('start > chromStart and end < chromEnd')
```